src/nvim/tui/tui.c

```diff
@@ -202,18 +126,94 @@ static void tui_stop(UI *ui)
   }
   xfree(data->write_loop);
   unibi_destroy(data->ut);
-  destroy_screen(data);
+  ugrid_free(&data->grid);
+  data->stop = true;
```

Comment:
This "stop signal" does not reaches `tui_main()` before `fg` is used, because `tui_main()` should still be polling for input.